### PR TITLE
Add `{:nilify, columns}` to `on_delete` reference option.

### PR DESCRIPTION
**Pull request by greg-rychlewski**
Postgres 15 allows the user to specify which columns should be set to null if the referenced record is deleted.

A side note: I noticed that one of the common on delete options from postgres/mysql is not present: `ON DELETE SET DEFAULT`. I was wondering if this was on purpose or I can submit a PR to add it? I wasn't sure if it was left out on purpose to avoid having to do the same thing for associations.

### CODE REVIEW

**Review by v0idpwn — APPROVED** [review]
Approved, just a small comment

File: lib/ecto/migration.ex [hunk 1]
```diff
@@ -1303,7 +1305,7 @@ defmodule Ecto.Migration do
     opts = Keyword.merge(foreign_key_repo_opts(), opts)
     reference = struct(%Reference{table: table}, opts)
 
-    unless reference.on_delete in [:nothing, :delete_all, :nilify_all, :restrict] do
+    unless valid_on_delete?(reference.on_delete) do
       raise ArgumentError, "unknown :on_delete value: #{inspect reference.on_delete}"
```
[the remark this answers is not in the document]
Comment (greg-rychlewski) [reply lib/ecto/migration.ex:1309]: Would it be better if there was a special message for `{:nilify, not_at_atom_list}` saying it expects the column list to be all atoms?

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
> A side note: I noticed that one of the common on delete options from postgres/mysql is not present: ON DELETE SET DEFAULT.

Sounds good to support it. At the same time nobody asked for it, so we are probably fine. :) Especially as nilifying likely covers the exact same cases.

**Review by v0idpwn — APPROVED** [review]

**Comment by greg-rychlewski** [comment]
>Sounds good to support it. At the same time nobody asked for it, so we are probably fine. :) Especially as nilifying likely covers the exact same cases.

Sorry I should have put it in description, this person here asked for `ON DELETE SET NULL columns` and `ON DELETE SET DEFAULT columns`:

https://groups.google.com/g/elixir-ecto/c/IIbzAgZYgXw

**Pull request merged**